omc3/model/accelerators/lhc.py

```diff
@@ -486,12 +488,43 @@ def _get_madx_main_sequence_loading(self) -> str:
                 "has to be specified (--lhcmode option missing?)."
             )
 
-    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str]) -> str:
+    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str], variable_categories: Sequence[str]) -> str:
         madx_script = self.get_base_madx_script()
         for corr_file in corr_files:
             madx_script += f"call, file = '{str(corr_file)}';\n"
-        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {self.dpp});\n"
+        
+        # Set the dpp in the twiss to the one in the model
+        dpp = self.dpp
+        if DELTAP_NAME in variable_categories: # If the dpp is a variable in the corrections, do twiss, correct, match
+            madx_script += self.get_update_deltap_script()
+            dpp = DELTAP_NAME # Set the dpp to the variable name (this seems the best way to include the update in the corrector script)
+        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {dpp});\n"
         return madx_script
+    
+    def get_update_deltap_script(self) -> str:
+        if self._uses_run3_macros():
+            knobx = f"dQx.b{self.beam}_op"
+            knoby = f"dQy.b{self.beam}_op"
+        elif int(self.year) == 2018: # This is very annoying - is there another way?
+            knobx = f"dQx.b{self.beam}"
+            knoby = f"dQy.b{self.beam}"
+        else:
+            knobx = f"KQTD.B{self.beam}"
+            knoby = f"KQTF.B{self.beam}"
+        return f"""
+{DELTAP_NAME} = {DELTAP_NAME}{self.dpp:+.15e};
```

Comment:
I cannot do this for MAD-X reasons. It will complain if self.dpp < 0:
`a = a + -2` errors in MAD-X

Reply:
Sigh... if you feel brave enough, include a small comment that this is same as a = a + value but works with negative values. Otherwise just resolve this comment and call it a day